crates/types/src/config/worker.rs

```diff
@@ -355,7 +358,19 @@ impl Default for StorageOptions {
 #[cfg_attr(feature = "schemars", schemars(rename = "SnapshotsOptions", default))]
 #[serde(rename_all = "kebab-case")]
 #[builder(default)]
-pub struct SnapshotsOptions {}
+pub struct SnapshotsOptions {
+    /// # Records per snapshot
+    ///
+    /// Number of log records after which a snapshot will be created.
+    ///
+    /// As snapshots are created asynchronously, the actual number of new records that will trigger
+    /// a snapshot will vary. The counter for the subsequent snapshot begins from the LSN at which
+    /// the previous snapshot export was initiated. Only  leader Partition Processors will take
+    /// snapshots for a given partition.
+    ///
+    /// Default: `None`
+    pub records_per_snapshot: Option<u64>,
```

Comment:
Maybe `Option<NonZeroU64>`?

Reply:
The default is to not do automatic snapshots (though it can be requested as normal via RPC / CLI). Perhaps we can for better ergonomics we might default this based on the type of loglet used - I feel that it's redundant and maybe a little surprising to leave snapshots on disk in a local-only single-node installation.